stdlib/public/Darwin/Foundation/Foundation.swift

```diff
@@ -29,7 +29,152 @@ public let NSNotFound: Int = .max
+///
+/// - returns: A localized version of the string designated by `key` in the
+///   table identified by `tableName`. If the localized string for `key` cannot
+///   be found within the table, the following semantics are used to determine
+///   returned value:
+///     - If `value` is the empty string, `key` is returned.
+///     - If `value` is not the empty string, `value` is returned.
+///
+/// Exporting Localizations
+/// -----------------------
+///
+/// Xcode can read through a project's code to find invocations of
+/// `NSLocalizedString(_:tableName:bundle:value:comment:)` and automatically
+/// generate the appropriate strings files for the project's base localization.
+///
+/// In Xcode, open the project file and, in the `Edit` menu, select
+/// `Export for Localization`. This will generate an XLIFF bundle containing
+/// strings files derived from your code along with other localizable assets.
+/// `xcodebuild` can also be used to generate the localization bundle from the
+/// command line with the `exportLocalizations` option.
+///
+///     xcodebuild -exportLocalizations -project <projectname>.xcodeproj \
+///                                     -localizationPath <path>
+///
+/// These bundles can be sent to translators for localization, and then
+/// reimported into your Xcode project. In Xcode, open the project file. In the
+/// `Edit` menu, select `Import Localizations...`, and select the XLIFF
+/// folder to import. You can also use `xcodebuild` to import localizations with
+/// the `importLocalizations` option.
+///
+///     xcodebuild -importLocalizations -project <projectname>.xcodeproj \
+///                                     -localizationPath <path>
+///
+/// Keys and Values
+/// ---------------
+///
+/// Words can often have multiple different meanings depending on the context
+/// in which they're used. For example, the word "Book" can be used as a noun—a
+/// printed literary work—and it can be used as a verb—the action of making a
+/// reservation. Words with different meanings which share the same spelling are
+/// heteronyms.
+///
+/// Different languages often have different heteronyms. "Book" in English is
+/// one such heteronym, but that's not so in French, where the noun translates
+/// to "Livre", and the verb translates to "Réserver". For this reason, it's
+/// important make sure that each use of the same phrase is translated
+/// appropriately for its context by assigning unique keys to each phrase and
+/// adding a description comment describing how that phrase is used.
+///
+///     NSLocalizedString("book-tag-title", value: "Book", comment: """
+///     noun: A label attached to literary items in the library.
+///     """)
+///
+///     NSLocalizedString("book-button-title", value: "Book", comment: """
+///     verb: Title of the button that makes a reservation.
+///     """)
+///
+/// Limitations
+/// -----------
+///
+/// Variables and interpolated strings cannot be passed into `key`, `tableName`,
```

Comment:
Can you phrase this in terms of you should do? i.e. "Only pass string literals as the `key`, ..." After that, you can mention how Xcode doesn't evaluate string interpolations, so translators/int'l users won't see the right thing, and how to use format strings.